### ISSUE TITLE
Incorrect table export, invisible foreign keys (4.5.5.1 & 4.6.0-rc2)

### ISSUE THREAD

**Issue by gugglegum** [phpmyadmin/phpmyadmin#12095]
Possible duplicated, but I believe my explanation is more detailed.

### Steps to reproduce
1) Create table from this SQL-dump:

```
CREATE TABLE `test` (
  `charterCompany` varchar(64) COLLATE utf8_unicode_ci DEFAULT NULL COMMENT 'Charter company',
  `status` int(10) unsigned NOT NULL DEFAULT '0' COMMENT 'Status',
  `partnerId` int(10) unsigned DEFAULT NULL COMMENT 'Partner',
  KEY `status` (`status`),
  KEY `partnerId` (`partnerId`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
```

2) Try to export this table

### Expected behaviour
It's should be correct SQL-dump.

### Actual behaviour
It generates incomplete invalid SQL dump, which cannot be imported (syntax error):

```
CREATE TABLE `test` (
  `charterCompany` varchar(64) COLLATE utf8_unicode_ci DEFAULT NULL COMMENT AS `Charter company`,
  `status` int(10) UNSIGNED NOT NULL DEFAULT '0'COMMENT
) ;
```

As you can see, PMA uses backtick quotes like "\`...\`" for comment string value and `AS` word which is not applicable here. The next line has word `COMMENT` without space separator and no expected string value after word `COMMENT`. 3rd column `partnerId` is absent. No keys exported.

If such table has foreign keys, they are not shown on page `tbl_relation.php` as they are absent. And these foreign keys are also missing in export dump.

### Server configuration
**Operating system**:
Win7 x64 SP1

**Web server:**
nginx

**Database:**
5.5.40-MariaDB

**PHP version:**
7.0.4

**phpMyAdmin version:**
4.5.5.1 & 4.6.0-rc2

### Client configuration
**Browser:**
Google Chrome 48.0.2564.116 m

**Operating system:**
Win7 x64 SP1

This is most epic failed bug ever!

**Comment by ibennetch** [comment]
The comment portion seems to be a duplicate of https://github.com/phpmyadmin/phpmyadmin/issues/12088, https://github.com/phpmyadmin/phpmyadmin/issues/12076, and https://github.com/phpmyadmin/sql-parser/issues/39 which will be fixed with 4.6.0 (but is not yet reflected in -rc2).

**Comment by ibennetch** [comment]
I don't see the foreign key problem; can you produce a small SQL file demonstrating the problem?

For instance, if I go in `sakila` to the `city` table, I can see the foreign keys as expected.

**Comment by gugglegum** [comment]
@ibennetch let's say we have this 2 tables:

```
CREATE TABLE `orders` (
  `charterCompany` varchar(64) COLLATE utf8_unicode_ci DEFAULT NULL COMMENT 'Charter company',
  `status` int(10) unsigned NOT NULL DEFAULT '0' COMMENT 'Status',
  `partnerId` int(10) unsigned DEFAULT NULL COMMENT 'Partner',
  KEY `status` (`status`),
  KEY `partnerId` (`partnerId`),
  CONSTRAINT `orders_partnerId` FOREIGN KEY (`partnerId`) REFERENCES `partners` (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;

CREATE TABLE `partners` (
  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `name` varchar(64) COLLATE utf8_unicode_ci NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
```
Import them into database. As you can see, we have a foreign key in `orders` table with reference to `partners` table. Now go to `tbl_structure.php` of `orders` table and you'll see none of foreign keys. Then switch to older version (e.g. 4.5.15.5) and foreign key appears!

**Comment by codewizz** [comment]
@ibennetch @gugglegum  I can confirm this. It looks like it's column comment related bug, once column has a comment set then foreign key is not visible for that column. Just remove all comments from ``orders`` table (example above) and foreign keys will show as expected.

This problem persists in 4.7.0-dev version aswell.

**Comment by ibennetch** [comment]
Okay, thanks for the clarification. I do confirm seeing the issue with 4.6.0-rc2 and master (which is 4.7.0-dev), but it appears fixed in QA_4_6 (which will become 4.6.1). I'll try to track down some more details.

**Comment by ibennetch** [comment]
The fix in QA_4_6 appears to come from f76ddbe7ab43e96b33ab7041a4917d86dda5e987 (which makes sense, since that fixed some comment-related regressions), which was also ported to https://github.com/phpmyadmin/sql-parser/commit/ad0e75fa however a new release of sql-parser hasn't occurred since then, so that is why the problem continues to occur with master.

Unless I'm rather mistaken, 4.6.0 will include this fix when it's released (probably next week), and master will incorporate the fix when there's a new release tagged for the sql-parser library (which I'll suggest doing immediately so we can confirm the fix in master). You can check the pending 4.6.0 release "Maintenance branch for 4.6.x releases." at https://www.phpmyadmin.net/try/ if you'd like to confirm it yourself.

**Comment by codewizz** [comment]
@ibennetch yes, you are right, it looks like to be fixed in "Maintenance branch for 4.6.x releases.", hopefully a new release of sql-parser will be merged onto master shortly.

Cheers!

**Comment by ibennetch** [comment]
Great, thank you for the confirmations!
